codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -480,6 +483,58 @@ private static boolean contains(Http2Headers headers, CharSequence name) {
         return false;
     }
 
+    private static boolean isValidHeaderValue(CharSequence value) {
+        // Validate value to field-content rule.
+        //  field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
+        //  field-vchar    = VCHAR / obs-text
+        //  VCHAR          = %x21-7E ; visible (printing) characters
+        //  obs-text       = %x80-FF
+        //  SP             = %x20
+        //  HTAB           = %x09 ; horizontal tab
+        //  See: https://datatracker.ietf.org/doc/html/rfc7230#section-3.2
+        //  And: https://datatracker.ietf.org/doc/html/rfc5234#appendix-B.1
+        int length = value.length();
+        if (length == 0) {
+            return true;
+        }
+        if (value instanceof AsciiString) {
+            return isValidHeaderValueAsciiString((AsciiString) value);
+        }
+        return isValidHeaderValueCharSequence(value);
+    }
+
+    private static boolean isValidHeaderValueAsciiString(AsciiString value) {
+        final byte[] array = value.array();
+        final int start = value.arrayOffset();
+        byte b = array[start];
+        if (b < 0x21 || b == 0x7F) {
+            return false;
+        }
+        int length = value.length();
+        for (int i = start + 1; i < length; i++) {
+            b = array[i];
+            if (b < 0x20 && b != 0x09 || b == 0x7F) {
+                return false;
```

Comment:
IIUC, it will mark obs-text (`%x80-FF`) as invalid, while spec allow it. Consider: https://github.com/apple/servicetalk/blob/ca56585472e946e1bfe2f56ba7ceb2eac4f6653c/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyH2HeadersToHttpHeaders.java#L546-L549

Reply:
That's basically the same logic. It says all controls, except for horizontal tab are invalid, and that DEL is invalid.

Reply:
This is true for `isValidHeaderValueCharSequence` because it uses `char b = value.charAt(0)`. 

For `AsciiString`, you operate with `byte`, which is negative for obs-text (`%x80-FF`). Please add a test to make sure `byte[] bytes = {(byte) 0x80, (byte) 0xFF}` is a valid header value for both paths: CharSequence and AsciiString.
Also, consider testing all scenarios for both CharSequence and AsciiString.

Reply:
Right you are. The signed bytes tricked me.